### PR TITLE
enable e2e testing in vagrant for main branch

### PR DESCRIPTION
**Pull request by davejrt**
This PR enables e2e testing for every commit to master. Currently this is non-blocking where e2e testing will return true even if the build fails and will not show the status of the main branch as failing. In future this will be disabled once the tests themselves *not the infra* are trustworthy enough.

### CODE REVIEW

**Review by uwedeportivo — APPROVED** [review]

File: .buildkite/pipeline.e2e.yml [hunk 1]
```diff
@@ -9,11 +9,11 @@ env:
   VAGRANT_RUN_ENV: 'CI'
 
 steps:
-  - artifact_paths: ./puppeteer/*.png;./client/web/e2e.mp4;./client/web/ffmpeg.log
```
Comment (eseliger) [inline .buildkite/pipeline.e2e.yml:12]: this seems to be a merge fault?

Comment (davejrt) [reply .buildkite/pipeline.e2e.yml:12]: The following code will scp those files out of the vagrant box and onto the agent, where they get uploaded.

https://github.com/sourcegraph/sourcegraph/pull/14533/files#diff-ea20c95d197018f59ffbcd3e71bab1d6R20-R22

```vagrant scp "$box":/sourcegraph/puppeteer/*.png ../
vagrant scp "$box":/sourcegraph/e2e.mp4 ../
vagrant scp "$box":/sourcegraph/ffmpeg.log ../
```

Comment (eseliger) [reply .buildkite/pipeline.e2e.yml:12]: Cool, thanks for clarifying :)

Comment (davejrt) [reply .buildkite/pipeline.e2e.yml:12]: Thanks for pointing this out...it was still wrong :-) I think I might have got a too trigger happy reverting some other junk.

**Pull request merged**